pages/rooms/[roomId]/index.tsx

```diff
@@ -19,45 +20,21 @@ interface RoomPageProps {
 }
 
 const RoomPage: NextPage<RoomPageProps> = ({ musicData }) => {
-  const [openAddSongScreen, setOpenAddSongScreen] = useState(false);
-  const [openPlaylistScreen, setOpenPlaylistScreen] = useState(false);
+  const {
+    state: { playingMusicId, playList, openAddSongScreen, openPlaylistScreen },
+    actions,
+  } = useRoomStore();
```

Comment:
```suggestion
  const { state: { playingMusicId, playList }, actions } = useRoomStore();
```

openAddSongScreen, openPlaylistScreen와 해당 모달을 열고 닫는 상태 변경 액션은 다른 페이지나 다른 곳에서 바라봐야할 이유가 없어보여서 제거해야 된다고 생각해!

위 [Modal추가 PR](https://github.com/mash-up-kr/musily-web/pull/80/files#diff-50c0d10de93a3ee490d6e91cd13ee3fb6014aa66f68d8182b98fa822803535da)에서 추가된 Modal과 같이 묶어준다면 관련 상태 처리를 한 곳에 응집시킬 수 있어서 좋을 거 같아
그러면 아래처럼 사용하는 곳에서 파악이 가능할 거 같아


```typescript
      <Modal
        trigger={({ open }) => <IconButton iconName="star" onClick={open} />}
        modal={({ close }) => <AddSongScreen onClickBackButton={close} />}
      />
```

Modal을 사용하면 state, actions에서 openDialog와 관련된 해당 상태와 액션을 제거할 수 있을 거 같아

형도 이 방식이 괜찮으면 이렇게 해보자고 하고 싶은데
이 PR이 머지되면 내 PR이 Conflict가 생길 거 같아서
먼저 형 PR머지하면 Conflict해결할게

Reply:
오 이렇게 Modal 을 따로 분리하는 구조가 말해준대로 상태관리도 편해지고 좋은거같아!

궁금한게 있는데, 위처럼 변경했을때 close 버튼 이외에 modal 을 close 하고 싶은 경우에는 어떤식으로 사용할 수 있을까? 
(예를들어, AddSongScreen 에서,  곡 추가후 모달을 닫히게 하고 싶을때 )

Reply:
이런 식으로 되면 페이지에 어떤 컴포넌트가 있는지 페이지단에서 찾기 쉽게 되고
다른 사람이 이 페이지를 다시 보더라도 한번에 찾아 갈 수 있지 않을까 싶어!
그리고 해당 모달 상태에 대한 내용은 Modal에 넘길 수 있으니까 페이지 내에나 atom에 상태를 유지할 필요가 없어지고 관련 코드가 관련된 위치에 있을 수 있게 되서 좋지 않을까 싶어

```typescript
const Page = () => {
  return (
  <...>
    <S.IconWrapper>
      <Actions.NewMusic />
      <Actions.Emoji />
      <Actions.ChangeMood />
    </S.IconWrapper>
  </...>
  )
}

const Actions = {
  NewMusic: () => (
    <Spacer type="vertical" align="center" gap={8}>
      <Modal
        trigger={({ open }) => <IconButton iconName="star" onClick={open} />}
        modal={({ close }) => <AddSongScreen onClickBackButton={close} />}
      />
      <S.IconText>곡추가</S.IconText>
    </Spacer>
  ),
  Emoji: () => {
    return (
      <Spacer type="vertical" align="center" gap={8}>
        <IconButton iconName="union" />
        <S.IconText>좋아요</S.IconText>
      </Spacer>
    );
  },
  ChangeMood: () => {
    return (
      <Spacer type="vertical" align="center" gap={8}>
        <IconButton iconName="union" />
        <S.IconText>무드변경</S.IconText>
      </Spacer>
    );
  },
}
```

만약 button 내부에 모달임을 숨기고 싶으면 NewMusicModalButton이 되고
그안에 Modal을 사용해서 넣을 수도 있지 않을까 싶어

Reply:
오케이 알았어! 일단 이 PR은 머지 하고 다른 리뷰내용들도 반영해볼게 고마워 ~!